src/libfetchers/git.cc

```diff
@@ -630,7 +636,7 @@ struct GitInputScheme : InputScheme
                 // everything to ensure we get the rev.
                 Activity act(*logger, lvlTalkative, actUnknown, fmt("making temporary clone of '%s'", repoDir));
                 runProgram("git", true, { "-C", tmpDir, "fetch", "--quiet", "--force",
-                        "--update-head-ok", "--", repoDir, "refs/*:refs/*" }, {}, true);
+                        "--update-head-ok", "--", repoDir, "refs/*:refs/*", input.getRev()->gitRev() }, {}, true);
```

Comment:
Wouldn't this cause the fetch to fail when the remote's `uploadpack.allowReachableSHA1InWant` is disabled?

Reply:
I'm not sure. There actually seem to be a few related options for configuring a `git` server and I'm not sure which combinations of them support fetching by rev. From [here](https://git-scm.com/docs/git-fetch-pack):

> The remote heads to update from. This is relative to $GIT_DIR (e.g. "HEAD", "refs/heads/master"). When unspecified, update from all heads the remote side has.
> If the remote has enabled the options uploadpack.allowTipSHA1InWant, uploadpack.allowReachableSHA1InWant, or uploadpack.allowAnySHA1InWant, they may alternatively be 40-hex sha1s present on the remote.

 But, I don't think remote configuration matters, here. In this case (regardless of the user-provided URL corresponding to either of a network-accessed repository or filesystem-accessed repository) the `fetch`, here, is against the fetcher cache whose `config` doesn't specify (or even support, I think) any of these configuration values. Fetching by rev is supported according to my experimentation and the reproduction embedded in the commit message.

Reply:
@roberth care to respond to this?